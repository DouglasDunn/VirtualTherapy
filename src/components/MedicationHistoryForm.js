import React from 'react';

export default class MedicationHistoryForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      medication: '',
      dose: '',
      purpose: '',
      sideEffects: '',
      current: ''
    };
  }

  onMedicationChange = (e) => {
    const medication = e.target.value;
    this.setState(() => ({ medication }));
  };

  onDoseChange = (e) => {
    const dose = e.target.value;
    this.setState(() => ({ dose }));
  };

  onPurposeChange = (e) => {
    const purpose = e.target.value;
    this.setState(() => ({ purpose }));
  };

  onSideEffetsChange = (e) => {
    const sideEffects = e.target.value;
    this.setState(() => ({ sideEffects }));
  };

  onCurrentChange = (e) => {
    const current = e.target.value;
    this.setState(() => ({ current }));
  };

  onSubmit = (e) => {
    e.preventDefault();

    this.props.onSubmit({
      medication: this.state.medication,
      dose: this.state.dose,
      purpose: this.state.purpose,
      sideEffects: this.state.sideEffects,
      current: this.state.current
    });
  };

  render() {
    return (
      <form className="form" onSubmit={this.onSubmit}>
        <input
          type="text"
          placeholder="Medication"
          autoFocus
          className="text-input"
          value={this.state.medication}
          onChange={this.onMedicationChange}
        />
        <input
          type="text"
          placeholder="Dose"
          className="text-input"
          value={this.state.dose}
          onChange={this.onDoseChange}
        />
        <textarea
          placeholder="Purpose"
          className="textarea"
          value={this.state.purpose}
          onChange={this.onPurposeChange}
        >
        </textarea>
        <textarea
          placeholder="Side Effects"
          className="textarea"
          value={this.state.sideEffects}
          onChange={this.onSideEffetsChange}
        >
        </textarea>
        <textarea
          placeholder="Current? If no, why not?"
          className="textarea"
          value={this.state.current}
          onChange={this.onCurrentChange}
        >
        </textarea>
        <div>
          <button className="button">Submit</button>
        </div>
      </form>
    )
  }
}
