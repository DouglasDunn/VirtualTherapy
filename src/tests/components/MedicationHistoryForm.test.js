import React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment';
import MedicationHistoryForm from '../../components/MedicationHistoryForm';
import medicationHistories from '../fixtures/medicationHistories';

test('should render MedicationHistoryForm correctly', () => {
  const wrapper = shallow(<MedicationHistoryForm />);
  expect(wrapper).toMatchSnapshot();
});

test('should render MedicationHistoryForm correctly with medicationHistory data', () => {
  const wrapper = shallow(<MedicationHistoryForm medicationHistory={medicationHistories[1]} />);
  expect(wrapper).toMatchSnapshot();
});

// test('should render error for invalid form submission', () => {
//   const wrapper = shallow(<ExpenseForm />);
//   expect(wrapper).toMatchSnapshot();
//   wrapper.find('form').simulate('submit', {
//     preventDefault: () => { }
//   });
//   expect(wrapper.state('error').length).toBeGreaterThan(0);
//   expect(wrapper).toMatchSnapshot();
// });

test('should set medication on input change', () => {
  const value = 'New medication';
  const wrapper = shallow(<MedicationHistoryForm />);
  wrapper.find('input').at(0).simulate('change', {
    target: { value }
  });
  expect(wrapper.state('medication')).toBe(value);
});

test('should set dose on input change', () => {
  const value = 'New dose';
  const wrapper = shallow(<MedicationHistoryForm />);
  wrapper.find('input').at(1).simulate('change', {
    target: { value }
  });
  expect(wrapper.state('dose')).toBe(value);
});

test('should set purpose on textarea change', () => {
  const value = 'New purpose value';
  const wrapper = shallow(<MedicationHistoryForm />);
  wrapper.find('textarea').at(0).simulate('change', {
    target: { value }
  });
  expect(wrapper.state('purpose')).toBe(value);
});

test('should set sideEffects on textarea change', () => {
  const value = 'New sideEffects value';
  const wrapper = shallow(<MedicationHistoryForm />);
  wrapper.find('textarea').at(1).simulate('change', {
    target: { value }
  });
  expect(wrapper.state('sideEffects')).toBe(value);
});

test('should set current on textarea change', () => {
  const value = 'New current value';
  const wrapper = shallow(<MedicationHistoryForm />);
  wrapper.find('textarea').at(2).simulate('change', {
    target: { value }
  });
  expect(wrapper.state('current')).toBe(value);
});

// test('should set amount if valid input', () => {
//   const value = '23.50';
//   const wrapper = shallow(<ExpenseForm />);
//   wrapper.find('input').at(1).simulate('change', {
//     target: { value }
//   });
//   expect(wrapper.state('amount')).toBe(value);
// });
//
// test('should not set amount if invalid input', () => {
//   const value = '12.122';
//   const wrapper = shallow(<ExpenseForm />);
//   wrapper.find('input').at(1).simulate('change', {
//     target: { value }
//   });
//   expect(wrapper.state('amount')).toBe('');
// });

// test('should call onSubmit prop for valid form submission', () => {
//   const onSubmitSpy = jest.fn();
//   const wrapper = shallow(<ExpenseForm expense={expenses[0]} onSubmit={onSubmitSpy} />);
//   wrapper.find('form').simulate('submit', {
//     preventDefault: () => { }
//   });
//   expect(wrapper.state('error')).toBe('');
//   expect(onSubmitSpy).toHaveBeenLastCalledWith({
//     description: expenses[0].description,
//     amount: expenses[0].amount,
//     note: expenses[0].note,
//     createdAt: expenses[0].createdAt
//   });
// });

// test('should set new date on date change', () => {
//   const now = moment();
//   const wrapper = shallow(<ExpenseForm />);
//   wrapper.find('SingleDatePicker').prop('onDateChange')(now);
//   expect(wrapper.state('createdAt')).toEqual(now);
// });
//
// test('should set calendar focus on change', () => {
//   const focused = true;
//   const wrapper = shallow(<ExpenseForm />);
//   wrapper.find('SingleDatePicker').prop('onFocusChange')({ focused });
//   expect(wrapper.state('calendarFocused')).toBe(focused);
// });
