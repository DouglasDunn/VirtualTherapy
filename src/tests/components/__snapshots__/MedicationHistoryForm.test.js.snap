// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MedicationHistoryForm correctly 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    autoFocus={true}
    className="text-input"
    onChange={[Function]}
    placeholder="Medication"
    type="text"
    value=""
  />
  <input
    className="text-input"
    onChange={[Function]}
    placeholder="Dose"
    type="text"
    value=""
  />
  <textarea
    className="textarea"
    onChange={[Function]}
    placeholder="Purpose"
    value=""
  />
  <textarea
    className="textarea"
    onChange={[Function]}
    placeholder="Side Effects"
    value=""
  />
  <textarea
    className="textarea"
    onChange={[Function]}
    placeholder="Current? If no, why not?"
    value=""
  />
  <div>
    <button
      className="button"
    >
      Submit
    </button>
  </div>
</form>
`;

exports[`should render MedicationHistoryForm correctly with medicationHistory data 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    autoFocus={true}
    className="text-input"
    onChange={[Function]}
    placeholder="Medication"
    type="text"
    value=""
  />
  <input
    className="text-input"
    onChange={[Function]}
    placeholder="Dose"
    type="text"
    value=""
  />
  <textarea
    className="textarea"
    onChange={[Function]}
    placeholder="Purpose"
    value=""
  />
  <textarea
    className="textarea"
    onChange={[Function]}
    placeholder="Side Effects"
    value=""
  />
  <textarea
    className="textarea"
    onChange={[Function]}
    placeholder="Current? If no, why not?"
    value=""
  />
  <div>
    <button
      className="button"
    >
      Submit
    </button>
  </div>
</form>
`;
